plugins/modules/rudder_server_settings.py

```diff
@@ -73,129 +72,144 @@
+            self.rudder_url = 'https://localhost/rudder'
+            self.validate_certs = False
+        if module.params.get('rudder_token', None) is None:
+            with open('/var/rudder/run/api-token') as system_token:
+                token = system_token.read()
+        else:
+            token = module.params['rudder_token']
+        self.headers = {
+            'X-API-Token': token,
+            'Content-Type': 'application/json',
+        }
 
-    def _send_request(self, url, data=None, headers=None, method="GET"):
+    def _send_request(self, url, data=None, headers=None, method='GET'):
         if data is not None:
             data = json.dumps(data, sort_keys=True)
 
         if not headers:
             headers = []
 
-        full_url = "{rudder_url}{path}".format(
+        full_url = '{rudder_url}{path}'.format(
             rudder_url=self.rudder_url, path=url
         )
 
         resp, info = fetch_url(
             self._module, full_url, data=data, headers=headers, method=method
         )
 
-        status_code = info["status"]
+        status_code = info['status']
         if status_code == 404:
-            return None
+            self._module.fail_json(failed=True, msg='Not found')
         elif status_code == 401:
             self._module.fail_json(
                 failed=True,
                 msg="Unauthorized to perform action '{}' on '{}'".format(
                     method, full_url
                 ),
             )
-        elif status_code == 403:
-            self._module.fail_json(failed=True, msg="Permission Denied")
         elif status_code == 200:
             return self._module.from_json(resp.read())
         else:
             self._module.fail_json(
                 failed=True,
-                msg="Rudder API answered with HTTP {} details: {} ".format(
-                    status_code, info["msg"]
+                msg='Rudder API answered with HTTP {} details: {} '.format(
+                    status_code, info['msg']
                 ),
             )
 
     def get_SettingValue(self, name):
-        url = "/api/latest/settings/{name}".format(name=name)
-        response = self._send_request(url, headers=self.headers, method="GET")
-        VALUE = response.get("data")
-        return VALUE.get("settings").get(name)
+        url = '/api/latest/settings/{name}'.format(name=name)
+        response = self._send_request(url, headers=self.headers, method='GET')
+        return response['data']['settings'][name]
 
     def set_SettingValue(self, name, value):
-        url = "/api/latest/settings/{name}?value={value}".format(
-            name=name, value=value
-        )
-
-        return self._send_request(url, headers=self.headers, method="POST")
+        url = '/api/latest/settings/{name}'.format(name=name)
+        current_server_settings = self.get_SettingValue(name)
+        data = value
+
+        update = False
+
+        for i_settings in self.settings_to_set:
```

Comment:
Where is this defined? why not just check the given value for the name taken in param?